proof.go

```diff
@@ -164,82 +216,22 @@ func (proof Proof) VerifyNamespace(h hash.Hash, nID namespace.ID, leaves [][]byt
 	nIDLen := nID.Size()
 	nth := NewNmtHasher(h, nIDLen, proof.isMaxNamespaceIDIgnored)
 
-	// perform some consistency checks:
-	// check that the root is valid w.r.t the NMT hasher
-	if err := nth.ValidateNodeFormat(root); err != nil {
-		return false
-	}
-	// check that all the proof.nodes are valid w.r.t the NMT hasher
-	for _, node := range proof.nodes {
-		if err := nth.ValidateNodeFormat(node); err != nil {
-			return false
-		}
-	}
-
-	// if the proof is an absence proof, the leafHash must be valid w.r.t the NMT hasher
-	if proof.IsOfAbsence() {
-		if err := nth.ValidateNodeFormat(proof.leafHash); err != nil {
-			return false
-		}
-	}
-
-	isEmptyRange := proof.start == proof.end
-	if isEmptyRange {
-		if proof.IsEmptyProof() && len(leaves) == 0 {
-			rootMin := namespace.ID(MinNamespace(root, nIDLen))
-			rootMax := namespace.ID(MaxNamespace(root, nIDLen))
-			// empty proofs are always rejected unless 1) nID is outside the range of
-			// namespaces covered by the root 2) the root represents an empty tree, since
-			// it purports to cover the zero namespace but does not actually include
-			// any such nodes
-			if nID.Less(rootMin) || rootMax.Less(nID) {
-				return true
-			}
-			if bytes.Equal(root, nth.EmptyRoot()) {
-				return true
-			}
-			return false
-		}
-		// the proof range is empty, and invalid
-		return false
+	// if empty range proof, check that the proof is valid
+	if proof.start == proof.end {
```

Comment:
invalid condition. I guess we can assume as invalid in case proof.start == proof.end.

Reply:
as mentioned in the previous comment, have to revisit this.